File: 1691b_ss.cpp
/**
 * @file 1691b_ss.cpp
 * @author Ukeme Edet (ukemeedet2207@gmail.com)
 * @brief The solution to the problem Shoe Shuffling(https://codeforces.com/problemset/problem/1691/B)
 * @version 0.1
 * @date 2024-06-08
 *
 * @copyright Copyright (c) 2024
 *
 */

#include <bits/stdc++.h>

using namespace std;

/**
 * @brief The main function
 *
 * @return int
 */
int main(void)
{
	ios::sync_with_stdio(0);
	cin.tie(0);
	unsigned short t;
	unsigned int n, i, l;
	vector<unsigned int> s;
	map<unsigned int, queue<unsigned int>> aq;

	cin >> t;
	while (t--)
	{
		cin >> n;
		s.resize(n), i = 1;
		aq.clear();
		for (auto &si : s)
		{
			cin >> si;
			aq[si].push(i++);
		}
		i = 0;
		for (auto &aqi : aq)
		{
			if (aqi.second.size() < 2)
				break;
			i++;
			l = aqi.second.front();
			aqi.second.pop();
			aqi.second.push(l);
		}
		if (i != aq.size())
		{
			cout << "-1\n";
			continue;
		}
		for (auto si : s)
		{
			cout << aq[si].front() << ' ';
			aq[si].pop();
		}
		cout << '\n';
	}
	return (EXIT_SUCCESS);
}

File: 1691b_ss.py
"""
@file: 1691b_ss.py
@author: Ukeme Edet (ukemeedet2207@gmail.com)
@brief: The solution to the problem Shoe Shuffling(https://codeforces.com/problemset/problem/1691/B)
@version: 1.0
@date: 2025-02-02

@copyright: Copyright (c) 2025
"""

from collections import defaultdict
from sys import stdin, stdout


def input():
    return stdin.readline().strip()


def print(string):
    return stdout.write(str(string) + "\n")


def main():
    for _ in range(int(input())):
        n = int(input())
        s = input().split()
        sm = defaultdict(list)
        for i in range(n):
            sm[s[i]].append(i)
        if any(len(v) == 1 for v in sm.values()):
            print(-1)
            continue
        res = [0] * n
        for v in sm.values():
            for i in range(1, len(v)):
                res[v[i]] = v[i - 1] + 1
            res[v[0]] = v[-1] + 1
        print(" ".join(str(x) for x in res))


if __name__ == "__main__":
    main()
